netman/api/switch_session_api.py

```diff
@@ -55,8 +55,8 @@ def open_session(self, session_id, hostname):
 
         """
 
-        switch = self.resolve_switch(hostname)
-        session_id = self.sessions_manager.open_session(switch, session_id)
+        switch_descriptor = self._get_switch_descriptor_from_request_headers(hostname)
```

Comment:
I propose that this class still use resolve switch and the factory is not injected in the session manager